dao/src/main/java/org/thingsboard/server/dao/ota/BaseOtaPackageService.java

```diff
@@ -186,6 +186,7 @@ public PageData<OtaPackageInfo> findTenantOtaPackagesByTenantIdAndDeviceProfileI
     }
 
     @Override
+    @Transactional
```

Comment:
please, check if it returns the right exception on UI when you violate a constraint (before and after your changes).
It can return a proxy instead the actual removal from the database until the end of the Transaction. 
Please add two screenshots with before and after UI exceptions to the description

Reply:
I do not know what you want
deleteOtaPackage has only one transactional operation: otaPackageDao.removeById